infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SenderBufferManager.java

```diff
@@ -185,12 +186,11 @@ public LogReplicationEntryMsg processAcks() throws InterruptedException, Executi
         return cf;
     }
 
-    public CompletableFuture<LogReplicationEntryMsg> sendWithBuffering(LogReplicationEntryMsg message,
-                                                                    String metricName,
-                                                                    Tag replicationTag) {
+    public CompletableFuture<LogReplicationEntryMsg> sendWithBuffering(LogReplicationEntryMsg message, String metricName, Tag replicationTag) {
         LogReplicationEntryMetadataMsg metadata = overrideSyncSeqNum(
                 message.getMetadata(), snapshotSyncSequenceNumber++);
         LogReplicationEntryMsg newMessage = overrideMetadata(message, metadata);
+        pendingMessages.append(newMessage);
```

Comment:
This missing line could have caused many bugs..

Reply:
right, this took me hours to realize! And luckily because I had disabled the metrics to ease logs readability I was able to hit this. This was an error doing copy/paste with the RPC change but it would've been critical for LR if metrics were disabled.